### ISSUE TITLE
Support multiple commentstrings

### ISSUE THREAD

**Issue by rosetree** [tpope/vim-commentary#44]
It would be great when commentary could recognize multiple ways of how a comments can be done in a language. This should only work to comment lines out; commenting should use a single `commentstring`.

As an example see the following comment in C before calling commentary:

```C
// no comment
```

After using `gcc` on that line:

```C
no comment
```

After using `gcc` a second time:

```C
/* no comment */
```

At the moment, using `gcc` on the first comment ends in:

```C
/* // no comment */
```

**Comment by justb3a** [comment]
+1

**Comment by somini** [comment]
There's `comments`, which has a list of comment styles.

**Comment by noscript** [comment]
@somini did you mean to link to another plugin?

**Comment by somini** [comment]
No, `comments` is a Vim setting. This is for the configuration per filetype, not the implementation.
